src/transformers/pipelines/fill_mask.py

```diff
@@ -122,9 +128,19 @@ def postprocess(self, model_outputs, top_k=5, target_ids=None):
                 probs = probs[..., target_ids]
 
             values, predictions = probs.topk(top_k)
+        elif self.framework == "flax":
+            tokens = np.copy(input_ids)
+            masked_index = np.where(input_ids == self.tokenizer.mask_token_id)[0]
+            logits = outputs[0, masked_index.item(), :]
+            probs = jax.nn.softmax(logits, axis=0)
```

Comment:
if possible I would actually try to keep everything in numpy here to avoid automatically allocating TPU or GPU for the post-processing. I think numpy as both faster and safer here actually (especially for sorting stuff like topk) cc @patil-suraj

Reply:
Waiting for @patil-suraj 's confirmation on this.

In PT the difference should be pretty negligible in most cases, but I think doing the softmax on GPU *should* (afaiu) be faster in pipe mode since, you can keep the workload on the GPU, meaning less CPU work, meaning less downtime for the GPU (since `postprocess` happens on the main thread).

This would need some proved numbers IMO (for the pure optimization aspect)